src/patches/lambda.js

```diff
@@ -44,11 +44,21 @@
       done(...args);
     };
 
+    let returned = null;
+
     try {
       addAllEventHandlers(eventHandlers);
-      return handler(event, context, finish);
+      returned = handler(event, context, finish);
     } catch (error) {
       finish(error);
+      return;
+    }
+
+    if (returned && typeof returned.then === 'function') {
```

Comment:
If the result of handler is analyzed and also `finish` provided to the handler, then it might be safest to use _.once() or something to protect against the callback being called multiple times. Even better would be to log if it is called multiple times.

Reply:
My understanding is that the Lambda runtime already protects against multiple invocations of the callback and I believe the `finish()` function is otherwise safe to call multiple times. However, I agree that logging would be nice. Since the logging applies equally well to the pure callback style and the promise style handlers I would like to handle the logging as a separate PR if you don't mind.